unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -3018,14 +3008,9 @@ public struct WhatDidITouch {
             float scaleOverSeconds = 1.0f,
             bool forceAction = false
         ) {
-            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
-                errorMessage = "Object ID appears to be invalid.";
-                actionFinished(false);
-                return;
-            }
 
             // if object is in the scene and visible, assign it to 'target'
-            SimObjPhysics target = getInteractableSimObjectFromId(objectId, forceAction);
+            SimObjPhysics target = getTargetObject(objectId: objectId, forceAction: forceAction);
```

Comment:
Can we change the name of `getTargetObject`? Currently it's unclear what it does (`getInteractableSimObjectFromId` is, for example, much more obvious).

Reply:
Looking at the code a little more, it seems that `getTargetObject` is just a thin wrapper around `getInteractableSimObjectFromId` that actually replicates some of the behavior of `getInteractableSimObjectFromId`. I think these two functions can safely be merged into one.

Reply:
it seems `getInteractableSimObjectFromId` is not used anymore after these changes, so I will just remove it. And yes `getTargetObject` is meant to grab a target object by objectId. I'll rename `getTargetObject` to a more descriptive `getTargetObjectFromId` for clarity.

Reply:
I think even `getTargetObjectFromId` is not descriptive enough (what is a "target" object?). The code seems to suggest that we're only returning the object when it's interactable, thus it makes more sense to me to have `getTargetObject` be named something like `getInteractableSimObjectFromId`. Also I believe that `getInteractableSimObjectFromId` is used inside `getTargetObject` unless you've already refactored this.

Reply:
`getInteractableSimObjectFromId` has already been moved out of `getTargetObject/getTargetObjectFromId`. I found no other functions that made reference to `getInteractableSimObjectFromId`.

I think changing the name isn't necessarily needed because it doesn't only return objects when they are interactable, since passing forceAction can cause an object to be returned as long as its Id is in the scene. This is sort of just being picky on the semantics though, so I think changing `getTargetObject` to `getinteractableSimObjectFromId` is fine if that is more clear.

Reply:
all instances of this function have now been renamed to `getInteractableSimObjectFromId`